sktime/forecasting/compose/_pipeline.py

```diff
@@ -1583,10 +1583,14 @@ def _get_X_for_fcX(self, X):
             return None
 
         if ixx == "complement":
-            X_for_fcX = X.drop(columns=self.columns)
-            if X_for_fcX.shape[1] < 1:
+            if X is None:
                 return None
-            return X_for_fcX
+            all_columns = X.columns
+            if self.columns is not None:
+                complemented_columns = list(set(all_columns) - set(self.columns))
+            else:
```

Comment:
In my understanding, this case means `ixx == "complement"` and `self.columns=None`, can you please confirm?

If so, this case will not arise due to L1578.

> if self.columns is None and ixx == "complement":

Reply:
Question for @fkiraly (based on git blame), can we change L1578 to just this: `if self.columns is None`? If no columns are specified, i.e. all columns are assumed to be future-unknown, can we use exogenous features in forecaster_X?

Reply:
From my understanding, if the `self.columns is None` can be sufficient for all cases as we don't need to check either complemented values or there is no usage of exogenous features at all `X.columns` will be used.